src/main/java/com/scaler/lld/design/assignments/singleton/FileBasedConfigurationManagerImpl.java

```diff
@@ -2,49 +2,63 @@
 
 public class FileBasedConfigurationManagerImpl extends FileBasedConfigurationManager {
 
+
+    private static FileBasedConfigurationManagerImpl instance;
+
+    // Private constructor to enforce Singleton pattern
+    private FileBasedConfigurationManagerImpl() {
+    }
     @Override
     public String getConfiguration(String key) {
-        // TODO Auto-generated method stub
-        throw new UnsupportedOperationException("Unimplemented method 'getConfiguration'");
+
+        return properties.getProperty(key);
     }
 
     @Override
     public <T> T getConfiguration(String key, Class<T> type) {
-        // TODO Auto-generated method stub
-        throw new UnsupportedOperationException("Unimplemented method 'getConfiguration'");
+
+        String value = properties.getProperty(key);
+        if (value != null) {
+            return convert(value, type);
+        }
+        return null;
```

Comment:
[Minor] This is correct. An improvement here could be to use the `Optional` class to reduce the indentation level.

Reply:
Done the changes